aspnetcore/fundamentals/http-requests.md

```diff
@@ -0,0 +1,241 @@
+
+An interface and a reply are defined to represent the external API and its response:
+
+```csharp
+public interface IHelloClient
+{
+    [Get("/helloworld")]
+    Task<Reply> GetMessageAsync();
+}
+
+public class Reply
+{
+    public string Message { get; set; }
+}
+```
+
+A typed client can be added, using Refit to generate the implementation:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddHttpClient("hello", c =>
+    {
+        c.BaseAddress = new Uri("http://localhost:5000");
+    })
+    .AddTypedClient(c => Refit.RestService.For<IHelloClient>(c));
+
+    services.AddMvc();
+}
+```
+
+The defined interface can be consumed where necessary, with the implementation provided by DI and Refit:
+
+```csharp
+[ApiController]
+public class ValuesController : ControllerBase
+{
+    private readonly IHelloClient _client;
+
+    public ValuesController(IHelloClient client)
+    {
+        _client = client;
+    }
+    
+    [HttpGet("/")]
+    public async Task<ActionResult<Reply>> Index()
+    {
+        return await _client.GetMessageAsync();
+    }
+}
+```
+
+## Outgoing request middleware
+
+`HttpClient` already has the concept of delegating handlers that can be linked together for outgoing HTTP requests. The `IHttpClientFactory` makes registration of per-named clients more intuitive. It supports registration and chaining of multiple handlers to build an outgoing request middleware pipeline. Each of these handlers is able to perform work before and after the outgoing request. This pattern is similar to the inbound middleware pipeline in ASP.NET Core. The pattern provides a mechanism to manage cross-cutting concerns around HTTP requests, including caching, error handling, serialization, and logging.
+
+To create a handler, define a class deriving from `DelegatingHandler`. Override the `SendAsync` method to execute code before passing the request to the next handler in the pipeline:
+
+[!code-csharp[Main](http-requests/samples/Handlers/ValidateHeaderHandler.cs?name=snippet1)]
+
+The preceding code defines a basic handler. It checks to see if an X-API-KEY header has been included on the request. If the header is missing it can avoid the HTTP call and return a suitable response.
+
+During registration, one or more handlers can be added to the configuration for a `HttpClient`. This task is accomplished via extension methods on the `IHttpClientBuilder`.
+
+[!code-csharp[](http-requests/samples/Startup.cs?name=snippet5)]
+
+In the preceding code, the `ValidateHeaderHandler` is registered as a transient service with DI. Once registered, `AddHttpMessageHandler` can be called, passing in the type for the handler.
```

Comment:
I think it's worth a callout here that a delegating handler **MUST** be registered as a transient.